lib/publiccloud/k8s_provider.pm

```diff
@@ -0,0 +1,78 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for amazon connection and authentication
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>, qa-c team <qa-c@suse.de>
+
+package publiccloud::k8s_provider;
+use Mojo::Base -base;
+use testapi;
+use mmapi 'get_current_job_id';
+
+has key_id => undef;
+has key_secret => undef;
+has region => undef;
+has resource_name => sub { get_var('PUBLIC_CLOUD_RESOURCE_NAME', 'openqa-vm') };
+has container_registry => undef;
+has provider_client => undef;
+has service => undef;
+
+sub init {
+    my ($self, $service) = @_;
+
+    die('The service must be specified') if (!$service);
+    $self->service($service);
+
+    my $provider = get_required_var('PUBLIC_CLOUD_PROVIDER');
+
+    if ($provider eq 'EC2') {
+        $self->provider_client(
+            publiccloud::aws_client->new(
+                key_id => $self->key_id,
+                key_secret => $self->key_secret,
+                region => $self->region,
+                service => $self->service
+            ));
+        $self->provider_client->init();
+    }
+    elsif ($provider eq 'GCE') {
+        # TODO
+    }
+    elsif ($provider eq 'AZURE') {
+        # TODO
+    }
+
+    $self->container_registry(get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REGISTRY", 'suse-qec-testing'));
+}
+
+=head2 get_container_registry_prefix
+Get the full registry prefix URL for any containers image registry of ECR based on the account and region
+=cut
+sub get_container_registry_prefix {
```

Comment:
This method seems to be called ONLY from `get_container_image_full_name`.
Wouldn't it be better to just have 1 method?

```
sub get_container_image_uri {
    my ($self) = @_;
    my $full_name_prefix = sprintf('%s.dkr.ecr.%s.amazonaws.com', $self->provider_client->aws_account_id, $self->region);
    return "$full_name_prefix/" . $self->container_registry . ":$tag";
}
```

Reply:
We use that in other places, for instance ecr.pm